diff --git a/modules/core/file/file.module b/modules/core/file/file.module
index 3d351fa..cb89bd9 100644
--- a/modules/core/file/file.module
+++ b/modules/core/file/file.module
@@ -6,7 +6,7 @@
  */
 
 // Load all Field module hooks for File.
-require_once DRUPAL_ROOT . '/modules/file/file.field.inc';
+require_once drupal_get_path('module', 'file') . '/file.field.inc';
 
 /**
  * Implements hook_help().
