Index: docroot/profiles/publisher/modules/custom/tve_adobe_pass/js/tve_adobe_pass.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- docroot/profiles/publisher/modules/custom/tve_adobe_pass/js/tve_adobe_pass.js	(date 1480065663000)
+++ docroot/profiles/publisher/modules/custom/tve_adobe_pass/js/tve_adobe_pass.js	(revision )
@@ -253,6 +253,19 @@
         // Onload callback.
         accessLoadedCheck
       );
+
+      swfobject.addLoadEvent(function() {
+        if (!checkIfRequestIsSent()) {
+          config.authNCheckedFailedCallback('flash');
+          stopAECheck();
+        }
+      });
+    }
+
+    function checkIfRequestIsSent() {
+      if (window.hasOwnProperty('performance') && 'getEntriesByName' in window.performance) {
+        return window.performance && window.performance.getEntriesByName(window.location.origin + config.adobePassAccessEnablerLoc).length != 0
+      }
     }
 
     /**
