diff --git a/modules/custom_p7/pub_workbench/pub_workbench.install b/modules/custom_p7/pub_workbench/pub_workbench.install
index 87f449dd8b354fc70eeb8871b274611e4c32db9f..4adacd7fe62ea0de8b64209da7ed1e46d158d399 100644
--- a/modules/custom_p7/pub_workbench/pub_workbench.install
+++ b/modules/custom_p7/pub_workbench/pub_workbench.install
@@ -22,7 +22,7 @@ function pub_workbench_install() {
       $operation->entity_type = 'node';
       $operation->entity_id = $record->nid;
       $operation->revision_id = $record->vid;
-      $operation->operation = 'workbench_moderation_publish';
+      $operation->operation = 'workbench_moderation_to_published';
       $operation->time_scheduled = $record->date;
       $operation->time_queued = 0;
       $operation->time_executed = 0;
